### DISCUSSION TITLE
non-Array stores with For

### DISCUSSION THREAD

**Discussion by georgfaust** [solidjs/solid#1418] [Q&A]
I have a store which is an object `{id: item, ...}`.
What is the right way to use For with it?

From the tutorial:

> The each prop expects an array, but you can turn other iterable objects into arrays with utilities like [Array.from](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from), [Object.keys](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys), or [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax).

But I'm interested in the `values`. This does not work (there is not re-rendering after a `setStore(id, key, value)`)
```
<For each={Object.values(store)}>
```
this works, but seems a little brutal:
```
<For each={Object.values(store).map((item) => ({ ...item }))}>
```

**Comment by Exelord** [comment]
In this example it works fine: https://playground.solidjs.com/anonymous/269bda8d-adcd-4b3e-a362-3f30a92c0be0

Could you try to reproduce it? It might be a case you are making a change on nested object which is not triggering the effect.
